test/cql-pytest/test_materialized_view.py

```diff
@@ -992,3 +992,209 @@ def test_base_clustering_prefix_deletion(cql, test_keyspace):
+        # The view partition key is a permutation of the base partition key.
+        with new_materialized_view(cql, table, '*', '(c,p),v', 'p is not null and c is not null and v is not null') as mv:
+            N = 1000
+            for i in range(N):
+                cql.execute(insert, [1, 10, i])
+                cql.execute(insert, [2, 20, i])
+
+            # Before the deletion, all N rows should exist in the base and the view
+            allN = list(range(N))
+            assert allN == [x.v for x in cql.execute(f"SELECT v FROM {table} WHERE p=1 AND c=10")]
+            assert allN == sorted([x.v for x in cql.execute(f"SELECT v FROM {mv} WHERE c=10 AND p=1")])
+            assert allN == [x.v for x in cql.execute(f"SELECT v FROM {table} WHERE p=2 AND c=20")]
+            assert allN == sorted([x.v for x in cql.execute(f"SELECT v FROM {mv} WHERE c=20 AND p=2")])
+
+            metrics_before = ScyllaMetrics.query(cql)
+            updates_before = metrics_before.get('scylla_database_total_view_updates_pushed_local')
+
+            cql.execute(f"DELETE FROM {table} WHERE p=1 AND c=10")
+
+            # After the deletion, all data should be gone from both base and view
+            assert [] == list(cql.execute(f"SELECT v FROM {table} WHERE p=1 AND c=10"))
+            assert [] == list(cql.execute(f"SELECT v FROM {mv} WHERE c=10 AND p=1"))
+            assert allN == [x.v for x in cql.execute(f"SELECT v FROM {table} WHERE p=2 AND c=20")]
+            assert allN == sorted([x.v for x in cql.execute(f"SELECT v FROM {mv} WHERE c=20 AND p=2")])
+
+            metrics_after = ScyllaMetrics.query(cql)
+            updates_after = metrics_after.get('scylla_database_total_view_updates_pushed_local')
+
+            print(f"scylla_database_total_view_updates_pushed_local: {updates_before} -> {updates_after}")
+            assert updates_after == updates_before + 1
+
+# Perform a batch operation, deleting a partition and also inserting a row
+# to that partition with a newer timestamp, and verify that the insertion
+# is not lost in the MV update.
+def test_base_partition_deletion_in_batch_with_insert(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, 'p int, c int, primary key (p,c)') as table:
+        insert = cql.prepare(f'INSERT INTO {table} (p,c) VALUES (?,?) USING TIMESTAMP 99')
+        with new_materialized_view(cql, table, '*', 'p,c', 'p is not null and c is not null') as mv:
+            cql.execute(insert, [0, 1])
+            cql.execute(insert, [0, 2])
+            cql.execute(insert, [0, 3])
+
+            # This should delete all the existing partition rows, and the new
+            # row insertion survives and remains the only row after the operation
+            # since it has the most recent timestamp.
+            cmd = 'BEGIN UNLOGGED BATCH '
+            cmd += f'DELETE FROM {table} USING TIMESTAMP 100 WHERE p=0; '
+            cmd += f'INSERT INTO {table} (p,c) VALUES (0,4) USING TIMESTAMP 101; '
```

Comment:
I recommend that you also add here an INSERT with TIMESTAMP 98 - i.e., before the partition tombstone, to see that this one does NOT survive.